From: Georges Basile Stavracas Neto <georges.stavracas@gmail.com>
Date: Fri, 3 Apr 2020 18:09:09 -0300
Subject: appDisplay: Set the folder icon geometry through CSS

The CSS engine is scale-aware, whereas simply setting the
width and height properties directly isn't.

Use CSS to set the folder icon.

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1176

Origin: upstream, 3.36.2, commit:1c27b68bcce305f2003c00a78bf965dbc1171136
---
 js/ui/appDisplay.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index e109b02..991ab2a 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -1376,12 +1376,12 @@ class FolderView extends BaseAppView {
         });
         layout.hookup_style(icon);
         let subSize = Math.floor(FOLDER_SUBICON_FRACTION * size);
-        let scale = St.ThemeContext.get_for_stage(global.stage).scale_factor;
 
         let numItems = this._orderedItems.length;
         let rtl = icon.get_text_direction() == Clutter.TextDirection.RTL;
         for (let i = 0; i < 4; i++) {
-            let bin = new St.Bin({ width: subSize * scale, height: subSize * scale });
+            const style = 'width: %dpx; height: %dpx;'.format(subSize, subSize);
+            let bin = new St.Bin({ style });
             if (i < numItems)
                 bin.child = this._orderedItems[i].app.create_icon_texture(subSize);
             layout.attach(bin, rtl ? (i + 1) % 2 : i % 2, Math.floor(i / 2), 1, 1);
